### PR TITLE
Add version flag to host agent

### LINKED ISSUE #278: Add --version flag to host agent binary
It will be useful to provide a `--version` flag in the host-agent binary as we will have multiple agent releases so that the user can know the version being used. The output could consist of information in the below format

```
$ byoh-hostagent-linux-amd64 --version
Version: v0.1.0_alpha.2
BuildDate: 
...
```

### PR DESCRIPTION
**Pull request by raunaksingwi**
**What this PR does / why we need it**:
Add version flag to the host agent so that the user can know the version being used.
The output shows the version of the binary and the time it was built.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #278 

**Additional information**
Used ldflags to embed the version number and time at build time in the binary.

The version number has to be passed as a variable to the Makefile. It will be passed by the GH action while the build is triggered.
When the version is not passed, it takes the default as "dev"

### CODE REVIEW

**Review by anusha94 — APPROVED** [review]
lgtm

File: agent/version/version_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+				BuildDate: version.BuildDate,
+				GoVersion: runtime.Version(),
+				Compiler:  runtime.Compiler,
+				Platform:  fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+			}
+			Expect(version.Get()).Should(Equal(expected))
+		})
+	})
+
+	Context("When version is set", func() {
```
Comment (dharmjit) [inline agent/version/version_test.go:60]: Suggestion: Should we have one more context under this for `when the git tag is not proper` for e.g. 1.0, etc.

**Review by dharmjit — COMMENTED** [review]
IMO, we need a consistent organization for `HostAgent` flags related code. Currently some of the code(help, labels flag) is inside `main` package and this PR introduces a new package `version`. How could we make it more cleaner? thoughts @anusha94 @raunaksingwi @huchen2021

**Comment by anusha94** [comment]
> IMO, we need a consistent organization for `HostAgent` flags related code. Currently some of the code(help, labels flag) is inside `main` package and this PR introduces a new package `version`. How could we make it more cleaner? thoughts @anusha94 @raunaksingwi @huchen2021

Agreed. All flags are all over the place now. @raunaksingwi Mind creating a new issue to address this so that this PR is unblocked?

**Review by anusha94 — APPROVED** [review]

**Review by dharmjit — APPROVED** [review]
lgtm

**Pull request merged**